### PR TITLE
Rolled-up additions to workflow and dataset interfaces

### PR DESCRIPTION
**Pull request by Robin-Wardle**
Caveats: these modifications were for my own use on the PYRAMID project at Newcastle University. Although I have used the implemented features, they lack comprehensive testing and documentation. I am aware that some debug `print()` statements are still present in the `Workflows` class.

### CODE REVIEW

File: dafni_cli/commands/execute.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+###############################################################################
+# Execute command
```
Comment (sarahbyrnie) [inline dafni_cli/commands/execute.py:2]: As discussed we don't currently want this execute ability so remove this file

File: dafni_cli/dafni.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from dafni_cli.commands.delete import delete
 from dafni_cli.commands.upload import upload
 from dafni_cli.commands.download import download
+from dafni_cli.commands.execute import execute
```
Comment (sarahbyrnie) [inline dafni_cli/dafni.py:7]: Remove this

File: dafni_cli/dafni.py [hunk 3]
```diff
@@ -18,6 +19,7 @@ def dafni():
 dafni.add_command(delete)
 dafni.add_command(upload)
 dafni.add_command(download)
+dafni.add_command(execute)
```
Comment (sarahbyrnie) [inline dafni_cli/dafni.py:22]: Remove this

File: dafni_cli/api/minio_api.py [hunk 4]
```diff
@@ -1,16 +1,18 @@
-from requests import Response
+import requests
 from pathlib import Path
-from typing import List
+from typing import Union, List
 
 from dafni_cli.consts import MINIO_UPLOAD_CT, DATA_UPLOAD_API_URL
+from dafni_cli.consts import DATA_DOWNLOAD_API_URL, DATA_DOWNLOAD_REDIRECT_API_URL
```
Comment (sarahbyrnie) [inline dafni_cli/api/minio_api.py:6]:
```suggestion
from dafni_cli.consts import (
    MINIO_UPLOAD_CT,
    DATA_UPLOAD_API_URL,
    DATA_DOWNLOAD_API_URL,
    DATA_DOWNLOAD_REDIRECT_API_URL,
```

File: dafni_cli/api/workflows_api.py [hunk 5]
```diff
@@ -103,7 +102,24 @@ def delete_workflow(jwt: str, workflow_version_id: str) -> Response:
...
     Args:
         jwt (str): JWT
-        workflow_version_id (str): model version ID for selected model
+        workflow_version_id (str): version ID of workflow to be deleted
     """
     url = WORKFLOWS_API_URL + "/workflows/" + workflow_version_id
     return dafni_delete_request(url, jwt)
+
+
+def execute_workflow(jwt: str, workflow_version_id: str, parameter_set_id: str = None):
```
Comment (sarahbyrnie) [inline dafni_cli/api/workflows_api.py:111]: Remove this function

**Review by sarahbyrnie — COMMENTED** [review]
Just comments on workflow execution as we discussed, looks good otherwise

**Review by joelvdavies — COMMENTED** [review]
I happy with these changes as well. There are some minor changes I would make but I can do those once merged.

**Comment by Robin-Wardle** [comment]
I've removed the Workflow execute facility and amended the minio_api formatting. The fixed model tests are also included now for you to review.

**Review by joelvdavies — APPROVED** [review]
Thanks for this. I am happy for it to be merged.

**Review by sarahbyrnie — APPROVED** [review]

**Pull request merged**